packages/test/test-end-to-end-tests/src/test/benchmark/DocumentCreator.ts

```diff
@@ -0,0 +1,55 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IContainer } from "@fluidframework/container-definitions";
+import { ChildLogger } from "@fluidframework/telemetry-utils";
+import { DocumentType, BenchmarkType } from "@fluidframework/test-version-utils";
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+import { ITestObjectProvider } from "@fluidframework/test-utils";
+import { DocumentMap } from "./DocumentMap";
+
+export interface IDocumentCreatorProps {
+	testName: string;
+	provider: ITestObjectProvider;
+	documentType: DocumentType;
+	benchmarkType: BenchmarkType;
+}
+
+export interface IDocumentProps extends IDocumentCreatorProps {
+	logger: ITelemetryLogger | undefined;
+}
+
+export interface IDocumentLoader {
+	initializeDocument(): Promise<void>;
+	loadDocument(): Promise<IContainer>;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+export class DocumentCreator {
```

Comment:
Seems unnecessary to have the class, as eslint points out. Can we just export a `createDocument()` function directly out of the module?

Reply:
yes, I was planning on evolving it but we can move it to a function for now :)